### PR TITLE
Handling  fetching token with claims on acquire token silent 

### PR DESCRIPTION
**Pull request by salman90**
## Purpose

*  After the claim challenge, I am adding the claim to acquireTokenSilent to fetch the correct from the cache.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* 1- The correct token is fetched from the cache
* 2- Claim challenge is stored and cleared from the local Storage on (POST, DELETE, PUT)

### CODE REVIEW

**Review by hectormmg — APPROVED** [review]
One suggestion, otherwise LGTM!

File: 6-AdvancedScenarios/3-call-api-acrs/SPA/src/pages/TodoList.jsx [hunk 1]
```diff
@@ -39,16 +39,10 @@ const TodoListContent = () => {
  * https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md
  */
 export const TodoList = () => {
```
Comment (sameerag) [inline 6-AdvancedScenarios/3-call-api-acrs/SPA/src/pages/TodoList.jsx:41]: Why is this removed?

Comment (salman90) [reply 6-AdvancedScenarios/3-call-api-acrs/SPA/src/pages/TodoList.jsx:41]: It was causing an automatic sign-in every time the user signs out

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1 [hunk 2]
```diff
@@ -193,194 +171,183 @@ Function ConfigureApplications
...
+    $dictionary = @{ "Enter_the_Application_Id_Here" = $clientAadApplication.AppId; "Enter_the_Tenant_Info_Here" = $tenantId; "Enter_the_Web_Api_Scope_here" = ("api://" + $serviceAadApplication.AppId + "/access_as_user") };
+    ReplaceInTextFile -configFilePath $configFile -dictionary $dictionary
+    Write-Host -ForegroundColor Green "------------------------------------------------------------------------------------------------" 
+    Write-Host "IMPORTANT: Please follow the instructions below to complete a few manual step(s) in the Azure portal":
+    Write-Host "- For service"
+    Write-Host "  - Navigate to Azure portal and set the 'accessTokenAcceptedVersion' to '2' in the application manifest" -ForegroundColor Red
+    Write-Host "  - Navigate to the API Permissions page and select 'Grant admin consent for (your tenant)" -ForegroundColor Red
+    Write-Host "  - Navigate to the Manifest page, find the 'optionalClaims' section and change its default value to request 'xms_cc' claims" -ForegroundColor Red  
+    Write-Host "- For client"
+    Write-Host "  - Navigate to Azure portal and set the 'replyUrlsWithType' to 'Spa' in the application manifest" -ForegroundColor Red
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1:341]: Since the code is stable now, lets use the new code gen to get rid of as many manual steps as possible

File: 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1 [hunk 3]
```diff
@@ -193,194 +171,183 @@ Function ConfigureApplications
...
 
-   # URL of the AAD application in the Azure portal
-   # Future? $clientPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   $clientPortalUrl = "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/CallAnAPI/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
-   Add-Content -Value "<tr><td>client</td><td>$currentAppId</td><td><a href='$clientPortalUrl'>msal-react-spa-acrs</a></td></tr>" -Path createdApps.html
+    # URL of the AAD application in the Azure portal
+    # Future? $clientPortalUrl = "https://portal.azure.com/#@"+$tenantName+"/blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/Overview/appId/"+$clientAadApplication.AppId+"/objectId/"+$clientAadApplication.ObjectId+"/isMSAApp/"
+    $clientPortalUrl = "https://portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationMenuBlade/CallAnAPI/appId/" + $clientAadApplication.AppId + "/objectId/" + $clientAadApplication.ObjectId + "/isMSAApp/"
```
Comment (kalyankrishna1) [inline 6-AdvancedScenarios/3-call-api-acrs/AppCreationScripts/Configure.ps1:300]: >$clientPortalUrl [](http://example.com/codeflow?start=4&length=16)

the earlier code used to print this url on console when manual steps were printed. This used to make the job of locating the app much easier in the portal. Could you check what happened..
